webview/src/webview_osx.mm

```diff
@@ -0,0 +1,389 @@
+{
+    CHECK_WEBVIEW_AND_RETURN();
+    CGRect screenRect = [[NSScreen mainScreen] visibleFrame];
+    g_WebView.m_WebViews[webview_id].frame = CGRectMake(x, y, width >= 0 ? width : screenRect.size.width, height >= 0 ? height : screenRect.size.height);
+    return 0;
+}
+
+dmExtension::Result Platform_AppInitialize(dmExtension::AppParams* params)
+{
+    g_WebView.Clear();
+    g_WebView.m_Mutex = dmMutex::New();
+    g_WebView.m_CmdQueue.SetCapacity(8);
+
+    return dmExtension::RESULT_OK;
+}
+
+dmExtension::Result Platform_AppFinalize(dmExtension::AppParams* params)
+{
+    dmMutex::Delete(g_WebView.m_Mutex);
+    return dmExtension::RESULT_OK;
+}
+
+dmExtension::Result Platform_Initialize(dmExtension::Params* params)
+{
+    return dmExtension::RESULT_OK;
+}
+
+dmExtension::Result Platform_Finalize(dmExtension::Params* params)
+{
+    for( int i = 0; i < dmWebView::MAX_NUM_WEBVIEWS; ++i )
+    {
+        if (g_WebView.m_WebViews[i]) {
+            DestroyWebView(i);
+        }
+    }
+
+    dmMutex::ScopedLock lk(g_WebView.m_Mutex);
+    for (uint32_t i=0; i != g_WebView.m_CmdQueue.Size(); ++i)
+    {
+        const Command& cmd = g_WebView.m_CmdQueue[i];
+        if (cmd.m_Url) {
+            free((void*)cmd.m_Url);
+        }
+    }
+    g_WebView.m_CmdQueue.SetSize(0);
+    return dmExtension::RESULT_OK;
+}
+
+
+dmExtension::Result Platform_Update(dmExtension::Params* params)
+{
+    if (g_WebView.m_CmdQueue.Empty())
+        return dmExtension::RESULT_OK; // avoid a lock (~300us on iPhone 4s)
+
+    dmMutex::ScopedLock lk(g_WebView.m_Mutex);
+    for (uint32_t i=0; i != g_WebView.m_CmdQueue.Size(); ++i)
+    {
+        const Command& cmd = g_WebView.m_CmdQueue[i];
+
+        dmWebView::CallbackInfo cbinfo;
+        switch (cmd.m_Type)
+        {
+        case CMD_EVAL_OK:
+            cbinfo.m_Info = &g_WebView.m_Info[cmd.m_WebViewID];
+            cbinfo.m_WebViewID = cmd.m_WebViewID;
+            cbinfo.m_RequestID = cmd.m_RequestID;
+            cbinfo.m_Url = 0;
+            cbinfo.m_Type = dmWebView::CALLBACK_RESULT_EVAL_OK;
+            cbinfo.m_Result = (const char*)cmd.m_Data;
+            RunCallback(&cbinfo);
+            break;
+
+        case CMD_EVAL_ERROR:
+            cbinfo.m_Info = &g_WebView.m_Info[cmd.m_WebViewID];
+            cbinfo.m_WebViewID = cmd.m_WebViewID;
+            cbinfo.m_RequestID = cmd.m_RequestID;
+            cbinfo.m_Url = 0;
+            cbinfo.m_Type = dmWebView::CALLBACK_RESULT_EVAL_ERROR;
+            cbinfo.m_Result = (const char*)cmd.m_Data;
+            RunCallback(&cbinfo);
+            break;
+
+        default:
+            assert(false);
+        }
+        if (cmd.m_Url) {
+            free((void*)cmd.m_Url);
+        }
+        if (cmd.m_Data) {
+            free(cmd.m_Data);
+        }
+    }
+    g_WebView.m_CmdQueue.SetSize(0);
+    return dmExtension::RESULT_OK;
+}
+
+#undef CHECK_WEBVIEW_AND_RETURN
+
+} // namespace dmWebView
+
+#endif // DM_PLATFORM_OSX
```

Comment:
It seems the actual differences between iOS and OSX are quite minimal?
Would it be feasible to use the same source file, and just use a few extra defines to include and call the correct apis?